### ISSUE TITLE
Crypt::TweetNacl test suite fails since 762c708fdcef

### ISSUE THREAD

**Issue by soundart** [rakudo/rakudo#2681]
## The Problem
Crypt::TweetNacl  test suite fails:
see https://travis-ci.org/soundart/perl6-tweetnacl/jobs/393657936
https://github.com/perl6/ecosystem-unbitrot/issues/368

AlexDaniel bisected this:

(01:25:19 PM) AlexDaniel: frankjh: 🥞🥞🥞 Bisecting Crypt::TweetNacl
(01:25:42 PM) AlexDaniel: frankjh: we'll see the result soon, but I guess you're right that it's a regression of some sort
(01:26:03 PM) AlexDaniel: frankjh: I confirm that on 2018.05 it passes OK
(01:27:14 PM) frankjh:  AlexDaniel:  travis uses now his is Rakudo version 2018.12-288-g5be7555fd built on MoarVM version 2018.12-45-ga9d02578aimplementing Perl 6.d. and it falis there too
(01:27:34 PM) AlexDaniel: frankjh: yeah… I wonder how we managed to miss that
(01:34:22 PM) AlexDaniel: frankjh: ok, it was bisected to this commit: https://github.com/rakudo/rakudo/commit/762c708fdcef3e52f49800ba9a621d29449e82fa
## Expected Behavior

pass

## Actual Behavior
```
t/01-basic.t .............. 
1..5
ok 1 - The object is-a 'NativeCall::Types::CArray[int8]'
ok 2 - The object is-a 'NativeCall::Types::CArray[int8]'
ok 3 - 
ok 4 - 
Cannot resolve caller STORE(Buf:D: NativeCall::Types::CArray[uint8]:D, :INITIALIZE); none of these signatures match:
    (Buf:D: Blob:D $blob, *%_)
    (Buf:D: int @values, *%_)
    (Buf:D: Iterable:D \iterable, *%_)
    (Blob:D: Iterable:D \iterable, :$INITIALIZE, *%_)
  in submethod BUILD at /home/travis/build/soundart/perl6-tweetnacl/lib/Crypt/TweetNacl/Basics.pm6 (Crypt::TweetNacl::Basics) line 35
  in block <unit> at t/01-basic.t line 17
# Looks like you planned 5 tests, but ran 4
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/5 subtests 
```

## Environment
* Operating system: 
* Compiler version (`perl6 -v`):

**Comment by AlexDaniel** [comment]
Ping @lizmat

**Comment by lizmat** [comment]
Fixed with ada116ef99 .  Since this was caught by tests, I guess we don't need any further tests?

**Comment by Kaiepi** [comment]
This is breaking tests on `IO::Socket::Async::SSL` because of this error:
```
> my CArray[CArray[uint32]] $a .= new: CArray[uint32].new(1),CArray[uint32].new(2)
CArray of CArray passed non-CArray object
  in method ASSIGN-POS at /home/morfent/Documents/perl6/rakudo/lib/NativeCall/Types.pm6 (NativeCall::Types) line 181
  in method new at /home/morfent/Documents/perl6/rakudo/lib/NativeCall/Types.pm6 (NativeCall::Types) line 235
  in method new at /home/morfent/Documents/perl6/rakudo/lib/NativeCall/Types.pm6 (NativeCall::Types) line 230
  in block <unit> at <unknown file> line 1
```

**Comment by jonathanstowe** [comment]
It also appears to be the culprit in  https://github.com/rakudo/rakudo/issues/2687

**Comment by lizmat** [comment]
Will look at this later tonight

**Comment by lizmat** [comment]
Sorry, it will be tomorrow.  Been too busy with too many other things to get the right mindset for this.

**Comment by lizmat** [comment]
Better fix with 5e5c909d1a

**Comment by lizmat** [comment]
Argh... it doesn't fix this one  :-(

**Comment by lizmat** [comment]
Removed original fix with 30fc5146ca to unbreak associated tickets

**Comment by lizmat** [comment]
Properly fixed with 84ed8548f97ff66c1ad8

**Comment by lizmat** [comment]
Tests added with b394b63c27c22bf6 after fixing the `Blob.new` case as well with 05b9aa0df802ab40b4
